### PR TITLE
Add support for Azure Linux 3.0

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to https://github.com/dotnet/dotnet-docker/issues/5331, https://github.com/dotnet/release/issues/901

Summary of changes:
1. Added Azure Linux 3.0 images
    1. 8.0: Additional supported distro
    2. 9.0: Replaced CBL-Mariner2.0
2. Updated Dockerfile templates with branding changes.  e.g. use azurelinux vs mariner terminology
3. Updated the tests accordingly
4. Added image size baseline entries for the new images but given the dev tooling doesn't support local updates, actual sizes will need to be updated after the PR is merged.

### CODE REVIEW

**Comment by lbussell** [comment]
Can you open a tracking issue in dotnet/release with the template ".NET Docker Release - New Linux Distro" for this? It contains a more detailed checklist.

**Comment by MichaelSimons** [comment]
> Can you open a tracking issue in dotnet/release with the template ".NET Docker Release - New Linux Distro" for this? It contains a more detailed checklist.

https://github.com/dotnet/release/issues/901 - That checklist is out of date - no Dockerfile templates.  I will update when done.

**Comment by MichaelSimons** [comment]
This is ready for review feedback.

File: tests/Microsoft.DotNet.Docker.Tests/OS.cs [hunk 1]
```diff
@@ -11,6 +11,13 @@ public class OS
         public const string Alpine318 = $"{Alpine}3.18";
         public const string Alpine319 = $"{Alpine}3.19";
 
+        // AzureLinux
+        public const string AzureLinux = "azurelinux";
+
+        public const string AzureLinuxDistroless = $"{AzureLinux}-distroless";
+        public const string AzureLinux30 = $"{AzureLinux}2.0";
```
Comment (mthalman) [inline tests/Microsoft.DotNet.Docker.Tests/OS.cs:18]:
```suggestion
        public const string AzureLinux30 = $"{AzureLinux}3.0";
```

**Review by mthalman — CHANGES_REQUESTED** [review]

**Review by mthalman — APPROVED** [review]

**Review by mthalman — APPROVED** [review]

**Review by jander-msft — APPROVED** [review]

**Pull request merged**